.github/workflows/model_training_tests_ci.yml

```diff
@@ -0,0 +1,64 @@
+name: Model training tests - CI
+run-name: Run tests defined in `model_training`
+
+on:
+  pull_request:
+    types: [opened, ready_for_review, synchronize, reopened]
+    branches:
+      - main
+      - develop
+  push:
+    branches:
+      - main
+      - develop
+
+defaults:
+  run:
+    working-directory: model_training
+
+jobs:
+  file-changes:
```

Comment:
I think we don't require a new job for this. It should work using [paths](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#onpushpull_requestpull_request_targetpathspaths-ignore) argument.

```yaml
on:
  pull_request:
    types: [opened, ready_for_review, synchronize, reopened]
    branches:
      - main
      - develop
     paths: 'model_training/**/*.py'
  push:
    branches:
      - main
      - develop
    paths: 'model_training/**/*.py'
```

Reply:
As in we don't need a new job to detect file change?